druid/src/widget/svg.rs

```diff
@@ -0,0 +1,222 @@
+                                let x1 = (x1 * scale) + offset.x;
+                                let y1 = (y1 * scale) + offset.y;
+                                let x2 = (x2 * scale) + offset.x;
+                                let y2 = (y2 * scale) + offset.y;
+                                let x = (x * scale) + offset.x;
+                                let y = (y * scale) + offset.y;
+
+                                //QUESTION is this order correct?
+                                path.curve_to((x1, y1), (x2, y2), (x, y));
+                            }
+                            PathSegment::ClosePath => {
+                                path.close_path();
+                            }
+                        }
+                    }
+                    match &p.fill {
+                        Some(fill) => {
+                            let brush = color_from_fill(fill);
+                            paint_ctx.fill(path.clone(), &brush);
+                        }
+                        None => {}
+                    }
+
+                    match &p.stroke {
+                        Some(stroke) => {
+                            let brush = color_from_stroke(stroke);
+                            paint_ctx.stroke(path.clone(), &brush, stroke.width.value());
+                        }
+                        None => {}
+                    }
+                }
+                NodeKind::Defs => {
+                    // TODO: what is this?
+                }
+                _ => {
+                    // TODO: handle more of the SVG spec.
+                    error!("{:?} is unimplemented", n.clone());
+                }
+            }
+        }
+    }
+}
+
+impl<T: Data> Widget<T> for SVG<T> {
+    fn paint(&mut self, paint_ctx: &mut PaintCtx, base_state: &BaseState, _data: &T, _env: &Env) {
+        //TODO: options for aspect ratio or scaling based on height
+        let scale = base_state.size().width / self.get_size().width;
+
+        let origin_x = (base_state.size().width - (self.get_size().width * scale)) / 2.0;
+        let origin_y = (base_state.size().height - (self.get_size().height * scale)) / 2.0;
+        let origin = Point::new(origin_x, origin_y);
+
+        self.svg_to_piet(scale, origin, paint_ctx);
+    }
+
+    fn layout(
+        &mut self,
+        _layout_ctx: &mut LayoutCtx,
+        bc: &BoxConstraints,
+        _data: &T,
+        _env: &Env,
+    ) -> Size {
+        bc.debug_check("SVG");
+
+        if bc.is_width_bounded() {
+            bc.max()
+        } else {
+            bc.constrain(self.get_size())
+        }
+    }
+
+    fn event(&mut self, _ctx: &mut EventCtx, _event: &Event, _data: &mut T, _env: &Env) {}
+
+    fn update(&mut self, _ctx: &mut UpdateCtx, _old_data: Option<&T>, _data: &T, _env: &Env) {}
+}
+
+fn color_from_fill(fill: &Fill) -> Color {
+    match fill.paint {
+        Paint::Color(c) => {
+            let alpha = (fill.opacity.value() * 255.0) as u8;
```

Comment:
This might be ever so slightly more idiomatic as `Color::rgb8(c.red, c.green, c.blue).with_alpha(fill.opacity.value())`. But that's entirely a style question, this is fine.